packages/material-ui/src/Zoom/Zoom.js

```diff
@@ -74,7 +76,7 @@ const Zoom = React.forwardRef(function Zoom(props, ref) {
   };
 
   return (
-    <Transition
+    <TransitionComponent
```

Comment:
Should we call the root component simply `Component`?

Reply:
Not a fan of the generic term if the component has a specific task.

Reply:
I wonder. It could feel surprising 🤔.

Reply:
That the component responsible for Transition is named TransitionComponent over Component? Is this seriously a concern?

Reply:
Would it be like we change the convention for the style rule, naming the root class name `transition` over `root`?

Reply:
What do you think of the idea to introduce a `Components?: { Root: React.ComponentType, … }` prop for v5 and empowering deep component customizations with styled component?

Similarly to what we do for the class names:

- `className` prop at the root
- `classes.root` object at the root
- `classes.x` object for the nested elements

we would have:

- `Component` prop at the root
- `Components.Root` object at the root
- `Components.x` object for the nested components